### PR TITLE
Integrate camera

### PR DESCRIPTION
**Pull request by AeroStun**
Latest [`smce-gd:vignore`](https://github.com/ItJustWorksTM/smce-gd/releases/tag/vignore) (and soon latest release) has camera support; this PR intends to integrate the camera to the app/car duo in this repository

### CODE REVIEW

**Comment by AeroStun** [comment]
Note: the camera library we polyfill is not supported on ESP32 (advertises Mbed instead), thus the CI does not build with it.

**Comment by platisd** [comment]
Oh wow, was this done over night? 😁
I could consider having a dummy camera class in Smartcar library instead. Let's discuss.

**Comment by AeroStun** [comment]
Ruthger and I did that over yesterday's afternoon and evening, and I just pushed until completion at night

File: arduino/smartcar_mqtt/smartcar_mqtt.ino [hunk 1]
```diff
@@ -44,14 +50,25 @@ void setup() {
...
+#endif
   }
+
   static unsigned long previousTransmission = 0;
   const auto currentTime = millis();
   if (currentTime - previousTransmission >= oneSecond) {
     previousTransmission = currentTime;
     const auto distance = String(front.getDistance());
     mqtt.publish("/smartcar/ultrasound/front", distance);
   }
```
Comment (platisd) [inline arduino/smartcar_mqtt/smartcar_mqtt.ino:72]: Thinking of it, these should be done (or rather the distance published), only if `mqtt` is connected.

File: arduino/smartcar_mqtt/smartcar_mqtt.ino [hunk 2]
```diff
@@ -20,14 +23,17 @@ const auto echoPin = 7;
 const auto maxDistance = 400;
 SR04 front(arduinoRuntime, triggerPin, echoPin, maxDistance);
 
+std::vector<char> frameBuffer;
```
Comment (platisd) [inline arduino/smartcar_mqtt/smartcar_mqtt.ino:26]: Isn't there a convention to use `unsigned char` when you want to use bytes until `std::byte` is made available?

File: arduino/smartcar_mqtt/smartcar_mqtt.ino [hunk 3]
```diff
@@ -1,5 +1,8 @@
 #include <Smartcar.h>
 #include <MQTT.h>
+#ifdef __SMCE__
+#include <OV767X.h>
+#endif
```
Comment (platisd) [inline arduino/smartcar_mqtt/smartcar_mqtt.ino:5]: This is my "issue".
This is a **very** idealistic camera but to be honest, it's how a camera API should be. I have used the particular library (or rather one for the specific sensors) and  to grab a frame you need to write a lot and ugly code from what I remember. :cry: 

Anyway, I don't have time to try it right now, but I suggest we include an "SMCE camera class" in the smartcar shield library instead, available only for the emulator.

**Review by platisd — COMMENTED** [review]
I haven't looked at everything yet

File: arduino/smartcar_mqtt/smartcar_mqtt.ino (hunk 2 above)
Comment (AeroStun) [reply arduino/smartcar_mqtt/smartcar_mqtt.ino:26]: In theory yes, in practice the MQTT message already stores `char`s

File: android/SmartcarMqttController/app/src/main/java/platis/solutions/smartcarmqttcontroller/MainActivity.java [hunk 4]
```diff
@@ -94,6 +100,23 @@ public void connectionLost(Throwable cause) {
                 @Override
                 public void messageArrived(String topic, MqttMessage message) throws Exception {
                     Log.i(TAG, "[MQTT] Topic: " + topic + " | Message: " + message.toString());
+                    if (topic.equals("/smartcar/camera")) {
+                        final ImageView cameraView = findViewById(R.id.imageView);
```
Comment (platisd) [inline android/SmartcarMqttController/app/src/main/java/platis/solutions/smartcarmqttcontroller/MainActivity.java:104]: Traditionally this has been placed in `onCreate`

**Review by platisd — COMMENTED** [review]
Still haven't tried it, just randomly nitpicking the commit

**Review by platisd — APPROVED** [review]

**Pull request merged**